open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,736 @@
+    }
+
+    /**
+     * Validates the presence of a specified key in a JSONArray (either the amount or the currency)
+     * in periodiclimits and checks if the associated value is a non-empty string.
+     *
+     * @param parentArray The JSONObject to be validated.
+     * @param key         The key to be checked for presence in the parentObj.
+     * @return true if the specified key is present in the parentObj and the associated value is a
+     * non-empty string.
+     */
+
+    public static JSONObject validateAmountCurrencyPeriodicLimits(JSONArray parentArray, String key) {
+        if (parentArray != null && key != null) {
+            // Check if the specified key is present in the parentArray
+            for (Object obj : parentArray) {
+                if (obj instanceof JSONObject) {
+                    JSONObject jsonObject = (JSONObject) obj;
+
+                    if (jsonObject.containsKey(key)) {
+                        Object value = jsonObject.get(key);
+
+                        // Check if the value associated with the key is a non-empty string
+                        if (value instanceof String && !((String) value).isEmpty()) {
+                            JSONObject validationResponse = new JSONObject();
+                            validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+                            return validationResponse; // Valid: The key is present, and the value is a non-empty String
+                        } else {
+                            String errorMessage = "Mandatory parameter '" + key + "' is not present in periodic" +
+                                    " limits or the value is not a string";
+                            return ConsentManageUtil.getValidationResponse(errorMessage);
+                            // Invalid: The value associated with the key is not a non-empty String
+                        }
+                    }
+                }
+            }
+        }
+        String errorMessage = "Mandatory parameter '" + key + "' of periodic limits is not present in payload";
+        return ConsentManageUtil.getValidationResponse(errorMessage);
+
+    }
+
+    /**
+     * Validates the consent initiation payload in the VRP request.
+     *
+     * @param request The JSONObject representing the VRP request.
+     * @return A JSONObject containing the validation response.
+     */
+    public static JSONObject validateConsentInitiation(JSONObject request) {
+
+        JSONObject validationResponse = new JSONObject();
+
+        JSONObject requestBody = (JSONObject) request;
+        JSONObject data = (JSONObject) requestBody.get(ConsentExtensionConstants.DATA);
+
+        //Validate initiation in the VRP payload
+        if (data.containsKey(ConsentExtensionConstants.INITIATION)) {
+
+            Object initiation = data.get(ConsentExtensionConstants.INITIATION);
+
+            String errorMessage = String.format(ErrorConstants.INVALID_PARAMETER_MESSAGE,
```

Comment:
Shall we refer https://github.com/wso2/financial-open-banking/pull/37/files#r1469084422

Reply:
Fixed in [e8d8048]